pkg/cri/server/sandbox_run.go

```diff
@@ -95,6 +84,40 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		},
 	)
 
+	// cleanupErr records the last error created by the critical cleanup in defer functions,
+	// like CNI teardown and stop running sandbox task.
+	// So when the cleanup is not completed for some reason, the CRI-plugin will remain the sandbox
+	// in the not-ready state, which can be cleanup by kubelet syncPod.
+	var cleanupErr error
+
+	defer func() {
+		if cleanupErr != nil {
+			log.G(ctx).WithError(cleanupErr).Errorf("there was error during resource cleanup and the sandbox will be put into store")
+
+			sandbox.Status = sandboxstore.StoreStatus(
+				sandboxstore.Status{
+					State: sandboxstore.StateNotReady,
```

Comment:
After check the code, the state should Unknown because the task is still alive. When the `StopPodSandbox` request is coming, the CRI will try to stop the task.

https://github.com/containerd/containerd/blob/main/pkg/cri/server/sandbox_stop.go#L78

Reply:
Thanks for catching this! We should use `Unknown` here. https://github.com/containerd/cri/pull/1037

Reply:
The task might be alive if the network setup succeeds. But if the network setup fails, the task doesn't even exist. Do we need to differentiate these two cases?

Reply:
> The task might be alive if the network setup succeeds. But if the network setup fails, the task doesn't even exist. Do we need to differentiate these two cases?

Even if the task is already gone, using unknown should be OK. `Unknown` is to make sure the sandbox is properly stopped before deletion.

Reply:
ok, I will update it to unknown